openshift/topics/advanced_concepts.adoc

```diff
@@ -594,6 +594,49 @@ There are two methods for passing the {project_name} adapter configuration to th
 
 See xref:Example-EAP-Manual[Example Workflow: Manually Configure an Application to Use {project_name} Authentication, Using SAML Client] for an end-to-end example of the manual {project_name} client registration method using a SAML client.
 
+=== Using {project_name} vault with Openshift secrets
+. Specify a directory for the vault using the *_SSO_VAULT_DIR_* environment variable. Add following snippets to your template: (Alternatively you can add *_SSO_VAULT_DIR_* variable to the environment in your deployment's configuration)
++
+[source,json,subs="attributes+,macros+"]
+----
+"parameters": [
+    ...
+    {
+        "displayName": "RH-SSO Vault Secret directory",
+        "description": "Path to the RH-SSO Vault directory.",
+        "name": "SSO_VAULT_DIR",
+        "value": "",
+        "required": false
+    }
+    ...
+]
+
+env: [
+    ...
+    {
+        "name": "SSO_VAULT_DIR",
+        "value": "${SSO_VAULT_DIR}"
+    }
+    ...
+]
+----
+
+. Create a secret in your openshift cluster:
++
+[source,bash,subs="attributes+,macros+"]
+----
+$ oc create secret generic rhsso-vault-secrets --from-literal=master_smtp__password=mySMTPPsswd
+----
+
+. Mount a volume to your deployment config using the `${SSO_VAULT_DIR}` as path. For already running deployment:
++
+[source,bash,subs="attributes+,macros+"]
+----
+oc set volume dc/sso --add --mount-path=${SSO_VAULT_DIR} --secret-name=rhsso-vault-secrets
+----
+
+. Now after a pod is created you can use specially crafted string within your {project_name} configuration. For example for using `mySMTPPsswd` created in this tutorial, you can use `${vault.smtp_password}` within `realm-name` realm as your smtp password and it will be replaced by `mySMTPPsswd` on demand.
```

Comment:
Please wrap the lines after ~ 80 chars to ease PR review

```suggestion
. Now after a pod is created you can use specially crafted string within your {project_name} configuration to refer to the secret. For example, for using `mySMTPPsswd` secret created in this tutorial, you can use `${vault.smtp_password}` within `realm-name` realm in the configuration of the smtp password and it will be replaced by `mySMTPPsswd` when used.
```